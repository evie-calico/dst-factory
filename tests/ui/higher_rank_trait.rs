use dst_factory::make_dst_factory;

trait MyTrait<'a> {}

#[make_dst_factory]
struct HigherRankTrait<'b> {
    id: i32,
    data: dyn for<'a> MyTrait<'b>,
}

fn main() {
}
