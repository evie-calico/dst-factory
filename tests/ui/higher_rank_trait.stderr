error: Higher-rank trait bounds (e.g., `for<'a> dyn Trait<'a>`) are not supported for the tail field.
 --> tests/ui/higher_rank_trait.rs:8:15
  |
8 |     data: dyn for<'a> MyTrait<'b>,
  |               ^^^^^^^^^^^^^^^^^^^
